internal/gps/source_manager.go

```diff
@@ -71,6 +72,9 @@ type SourceManager interface {
 	// project/source root.
 	DeduceProjectRoot(ip string) (ProjectRoot, error)
 
+	// SourceURLsForPath takes an import path and deduces the possible source URLs
```

Comment:
let's be a little more verbose and explanatory with this:

"SourceURLsForPath takes an import path and deduces the set of source URLs that may refer to a canonical upstream source. In general, these URLs differ only by protocol (e.g. https vs. ssh), not path."